pallets/corporate-actions/src/lib.rs

```diff
@@ -305,16 +308,31 @@ pub struct CAId {
 /// Weight abstraction for the corporate actions module.
 pub trait WeightInfo {
     fn set_max_details_length() -> Weight;
-
     fn reset_caa() -> Weight;
-    fn set_default_targets(num_targets: u32) -> Weight;
+    fn set_default_targets(i: u32) -> Weight;
     fn set_default_withholding_tax() -> Weight;
-    fn set_did_withholding_tax() -> Weight;
+    fn set_did_withholding_tax(existing_overrides: u32) -> Weight;
+
+    fn initiate_corporate_action_use_defaults(whts: u32, target_ids: u32) -> Weight;
+    fn initiate_corporate_action_provided(whts: u32, target_ids: u32) -> Weight;
+    fn initiate_corporate_action(whts: u32, target_ids: u32) -> Weight {
+        Self::initiate_corporate_action_use_defaults(1, 1)
```

Comment:
I am not sure if we should use the default implementation for any `WeightInfo` traits. If we change, in the future, the implementation of that extrinsic (i.e. an optimization for an specific use case), this weight function won't reflect that improvement and it could go unnoticed by the programmer. 
I mean this would be a relation between 2 weight fn, instead of a direct relation between an extrinsinc and its weight fn (1 to 1).